File: backend/src/main/java/br/com/guerreiro_dev/backend/domain/Veiculo.java
package br.com.guerreiro_dev.backend.domain;


import br.com.guerreiro_dev.backend.domain.enums.StatusVeiculo;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serial;
import java.io.Serializable;
import java.util.UUID;

@Getter
@Setter
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tb_veiculo")
public class Veiculo implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @EqualsAndHashCode.Include
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    private String placa;
    private String chassi;
    private String modelo;
    private String anoFabricacao;
    private String horimetroAtual;
    private String kmAtual;

    @Enumerated(EnumType.STRING)
    private StatusVeiculo status;



}

File: frontend/types/veiculo.ts
export interface Veiculo {
  id: string; // Agora ele é obrigatório na leitura (GET)
  placa: string;
  chassi: string;
  modelo: string;
  status: string;
  anoFabricacao: string;
  horimetroAtual: string;
  kmAtual: string;
}

export type VeiculoCreate = Omit<Veiculo, 'id'>;